### PR TITLE
fix(material/chips): remove button is too small

### PR DESCRIPTION
**Pull request by essjay05**
Fixes Angular Components Chip component where the touch target of the remove button of a chip is too small. Updates .mat-mdc-chip-remove to target more specific styles to override original style of padding: 8px so that the user has a larger touch target particularly on mobile devices.

Fixes b/286959517

Before [screenshot](https://screenshot.googleplex.com/LY4J6jk4b3TFyQo)
After [screenshot](https://screenshot.googleplex.com/5zpHzNya7CxhMnx)

BREAKING CHANGE: updates chip remove button touch target to increase accessibility of the button especially on touch/mobile devices.

### CODE REVIEW

File: src/material/chips/chip.scss [hunk 1]
```diff
@@ -398,6 +398,17 @@
   }
 }
 
+//Increases mdc-dhip-remove trailing icon action touch-target in order to fix b/286959517
```
Comment (mmalerba) [inline src/material/chips/chip.scss:401]: nit: space after `//`
nit: "dhip" => "chip"

Comment (essjay05) [reply src/material/chips/chip.scss]: Updated comment.

**Comment by essjay05** [comment]
As recommended, tested on Mac Chrome and Firefox against React, Vue, W3, and dialog native html. The issue is reproducible in each of these when using a Mac OS. 

Examples tested:
-[React dialog demo](https://headlessui.com/react/dialog)
-[Vue dialog demo](https://headlessui.com/v1/vue/dialog)
-[W3.org dialog demo](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/)
-[Native html dialog demo](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)

File: src/material/chips/chip.scss [hunk 2]
```diff
@@ -695,16 +695,21 @@ $_avatar-trailing-padding: 8px;
 
   // Used as a state overlay.
   &::after {
```
Comment (mmalerba) [inline src/material/chips/chip.scss:697]: This is better, but now the hover ripple is giant and off-center. I think you can undo the changes below and just add the following styles:
```
    box-sizing: border-box;
    padding: 12px; // should figure out how to calculate this, logically its the (touch-target-size - ripple-size)/2
    margin: -12px;
    background-clip: content-box;
```

Comment (essjay05) [reply src/material/chips/chip.scss:697]: Hi @mmalerba, to make sure the touch-target meets the accessible 48px x 48px should the padding be 24px instead and margin -24px?

Comment (essjay05) [reply src/material/chips/chip.scss:697]: Hi @mmalerba! I updated the changes based on your recommendation, but was wondering if I should change the padding to be 24px and margin -24px so the touch target is the accessible minimum 48px x 48px? Let me know your thoughts. Thanks for the consult!
[Latest commit](https://github.com/angular/components/pull/29351/commits/37d622067faeaf5f2e36cc142f9017d032f692e3)
[Updated screenshot](https://screenshot.googleplex.com/AiyY96F9SUYmacp)

Comment (mmalerba) [reply src/material/chips/chip.scss:697]: I think it should be the correct size if you add back the original styles you accidentally removed:
```
$offset: 2px;
top: 0 - $offset;
bottom: 0 - $offset;
left: 8px - $offset;
right: 8px - $offset;
border-radius: 50%;
```

Also, I notice that in some places we're setting `background: ...` for the `::after` element which clobbers the `background-clip` value. We should change those to `background-color` so they're not inadvertently changing the clip. We'll see what kind of issues we run into presubmitting. It's possible people are doing their own overrides of `background`, in which case I think we'd just have to apply `background-clip` with a higher specificity.

File: src/material/chips/chip.scss [hunk 3]
```diff
@@ -695,16 +695,20 @@ $_avatar-trailing-padding: 8px;
...
     display: block;
     opacity: 0;
     position: absolute;
-    top: 0 - $offset;
-    bottom: 0 - $offset;
-    left: 8px - $offset;
-    right: 8px - $offset;
-    border-radius: 50%;
+    // Increases touch target size to improve accessibility and fix b/286959517
+    z-index: 100;
```
Comment (mmalerba) [inline src/material/chips/chip.scss:705]: I don't think `z-index` or `pointer-events` is actually needed, I just put that in my original snippet to be extra safe

File: src/material/chips/chip.scss [hunk 4]
```diff
@@ -695,16 +695,20 @@ $_avatar-trailing-padding: 8px;
...
+    pointer-events: all;
+    box-sizing: border-box;
+    padding: calc(($_touch-target-size - $_ripple-size)/2);
+    margin: calc((($_touch-target-size - $_ripple-size)/2) * -1);
+    -webkit-background-clip: content-box;
```
Comment (mmalerba) [inline src/material/chips/chip.scss:710]: background-clip: content-box has good browser support, this shouldn't be needed. If you added it because of some lint check you can supress the lint. https://caniuse.com/mdn-css_properties_background-clip_content-box

File: src/material/chips/chip.scss (hunk 2 above)
Comment (essjay05) [reply src/material/chips/chip.scss:697]: Noted! Apologies for my confusion, returning the styles and updating the other `::after` styles as recommended to solve for background-clip.

File: src/material/chips/chip.scss (hunk 4 above)
Comment (essjay05) [reply src/material/chips/chip.scss]: Updated based on your recommendations and confirmed appropriate touch-target size: https://screenshot.googleplex.com/4wVGKZ6nV3s7cy6.

File: src/material/chips/chip.scss (hunk 3 above)
Comment (essjay05) [reply src/material/chips/chip.scss]: Removed

**Comment by mmalerba** [comment]
Looks good! I notice that the chip content seems slightly off center, but that appears to be an issue at HEAD, likely a regression introduced by https://github.com/angular/components/pull/29357 and not something blocking for this PR

**Review by mmalerba — APPROVED** [review]
